### PR TITLE
send idle ec2 instance notification

### PR DESCRIPTION
**Pull request by rajnikantsh**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3054)

### CODE REVIEW

**Review by REASY — APPROVED** [review]

**Review by JustinPihony — APPROVED** [review]
Nothing major - just a question to clarify - maybe improve in another PR?

**Comment by JustinPihony** [comment]
@rajnikantsh Any word on my comment - and/or when you plan on putting this live?

**Comment by JustinPihony** [comment]
test!

**Comment by rajnikantsh** [comment]
> @rajnikantsh Any word on my comment - and/or when you plan on putting this live?

@JustinPihony this is on live now

File: aws/src/main/python/cloudwatchAllInstances/lambda_function.py [hunk 1]
```diff
@@ -100,15 +103,40 @@ def get_running_instance_for(region):
...
+        responses = cloudtrail.lookup_events(LookupAttributes=[ { 'AttributeKey': 'ResourceName','AttributeValue': instance_id }],
+                                             StartTime=today - dateutil.relativedelta.relativedelta(months=1),
+                                             EndTime=today
+                                             )
+        for event in responses['Events']:
+            if event['EventName'] == 'RunInstances':
+                username=event['Username']
+
+    if username is not None:
+        user_email_ids=os.environ['USER_EMAIL_MAPPER']
```
[the remark this answers is not in the document]
Comment (rajnikantsh) [reply aws/src/main/python/cloudwatchAllInstances/lambda_function.py:135]: We need this email mapping in idleNotifier as well so i keep this mapping in lambda function

File: aws/src/main/python/idleNotifier/lambda_function.py [hunk 2]
```diff
@@ -24,13 +24,18 @@ def lambda_handler(event, context):
...
+    channel_id=safe_value_with_default(slack_ids_as_dict, email, 'channel')
+
     payload = {
         "blocks": [
             {
                 "type": "section",
                 "text": {
                     "type": "mrkdwn",
-                    "text": f"*EC2 Idle Alarm Triggered*\n> *Alarm Name*\n> {alarm_name}\n> *Instance Name*\n> {instance_name}\n> *Trigger Subject*\n> {subject}\n> *Link to Alarm*\n> https://console.aws.amazon.com/cloudwatch/home?region={region}#alarmsV2:alarm/{alarm_name}\n> *Link to Instance*\n> https://console.aws.amazon.com/ec2/home?region={region}#Instances:instanceId={instance}"
+                    "text": f"*<@{channel_id}|cal> EC2 Idle Alarm Triggered*\n> *Alarm Name*\n> {alarm_name}\n> *Instance Name*\n> {instance_name}\n> *Trigger Subject*\n> {subject}\n> *Link to Alarm*\n> https://console.aws.amazon.com/cloudwatch/home?region={region}#alarmsV2:alarm/{alarm_name}\n> *Link to Instance*\n> https://console.aws.amazon.com/ec2/home?region={region}#Instances:instanceId={instance}"
```
Comment (JustinPihony) [inline aws/src/main/python/idleNotifier/lambda_function.py:38]: Notes for myself after this has been running:
* @here is really !here - and sometimes this is a miss (still diagnosing)
* The email is ONLY filled in for when we send the JSON, not when it comes from SNS (Checking for a solution to store in the alarm - short of duplicating checking direct from the instance)

**Pull request merged**